### PR TITLE
Monitor and log CPU usage

### PR DESCRIPTION
**Pull request by lynxthecat**
Leverage /proc/stat to support monitoring and logging CPU use across detected cores.

### CODE REVIEW

**Comment by lynxthecat** [comment]
@rany2 I wonder if you see any issues or improvements to my code here?

File: cake-autorate.sh [hunk 1]
```diff
@@ -1495,6 +1533,42 @@ do
 				set_shaper_rate "dl"
 				set_shaper_rate "ul"
 
+				# update CPU load if CPU monitoring interval exceeded
+				if (( output_cpu_stats || output_cpu_raw_stats ))
+				then
+					if (( t_start_us > t_last_cpu_usage_check_us + monitor_cpu_usage_interval_us ))
```
Comment (rany2) [inline cake-autorate.sh:1539]: unnecessary nesting, could be combined into one if statement

**Review by rany2 — CHANGES_REQUESTED** [review]

**Comment by lynxthecat** [comment]
How about now?

**Review by rany2 — APPROVED** [review]
Not sure how useful these metrics would be, but we could easily undo it if it causes problems.

**Pull request merged**